### PR TITLE
Build Android targeting API 26 and test with API 24 system image.

### LINKED ISSUE #473: Switch from GCC to clang for android builds
Currently we use `arm-linux-androideabi-gcc` and `arm-linux-androideabi-g++` for Android builds. According to [the NDK 14 changelog](https://github.com/android-ndk/ndk/wiki/Changelog-r14), support for GCC is deprecated in the NDK. So, let's switch to clang.

### LINKED ISSUE #603: Travis CI Android emulator-based tests for Android are broken
It seems like the Android SDK changed in some non-backward-compatible way. If somebody has a fix, I'd take a PR for it. Otherwise I think I might try rewriting all the non-Windows CI stuff using Docker and then using the Docker-ified stuff, ETA in Feburary.

### PR DESCRIPTION
**Pull request by pietro**
Fixes #603 and we can also close #473 even though the switch to clang didn't happen in this pr.

### CODE REVIEW

**Comment by pietro** [comment]
@briansmith I need to look into being able to update the sdk tools and systems images. I won't have much time in the next few weeks though. Can you delete the caches for ring on travis and re-run the build?

**Comment by briansmith** [comment]
Thanks!

> Can you delete the caches for ring on travis and re-run the build?

I just did this.

File: README.md [hunk 1]
```diff
@@ -215,8 +215,8 @@ the table below. The C compilers listed are used for compiling the C portions.
 </tr>
 <tr><td>Android</td>
     <td>32&#8209;bit&nbsp;ARM</td>
-    <td>Built using the Android SDK 24.4.1 and Android NDK 14, tested using the
-        Android emulator.</td>
+    <td>Built using the Android SDK 24.4.1 and Android NDK 17 targeting API level 26,
+        tested using the Android emulator on API levle 24 system images..</td>
```
Comment (briansmith) [inline README.md]: s/levle/level/

**Comment by briansmith** [comment]
Thanks! This was merged as 49ec4809b6afd1c6461ecd470828ef31fb6af655. I clarified some of the documentation in 5c8c13e6d8984eed58d5f65ce17418bb37b934ba and 88d3279cba40caa22a554857a46c8ba89633a317; this included fixing the typo I mentioned above.

Thanks again!

**Pull request merged**